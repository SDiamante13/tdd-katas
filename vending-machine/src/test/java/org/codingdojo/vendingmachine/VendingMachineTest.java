package org.codingdojo.vendingmachine;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;

class VendingMachineTest {

    private VendingMachine machine;
    private Map<String, Integer> coins;

    @BeforeEach
    void setUp() {
        machine = new VendingMachine();
        coins = new HashMap<>() {{
            put("penny", 1);
            put("nickel", 5);
            put("dime", 10);
            put("quarter", 25);
        }};
    }

    @Test
    void test_accept_coins() {
        String initialState = new VendingMachinePrinter(machine).print();

        machine.insertCoin(coins.get("nickel"));

        String finalState = new VendingMachinePrinter(machine).print();

        Approvals.verify(initialState + "********************\n" + finalState);
    }
}
