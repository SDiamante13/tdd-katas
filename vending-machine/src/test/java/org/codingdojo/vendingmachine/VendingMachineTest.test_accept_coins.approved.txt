VendingMachine
Display                                          INSERT COIN
Balance                                                    0
Coins                                                     []
********************
VendingMachine
Display                                                $0.05
Balance                                                    5
Coins                                                    [5]
